apps/total-typescript/src/ability/abilitiy.test.ts

```diff
@@ -1,12 +1,37 @@
-import {AppAbility, defineRulesForPurchases} from './ability'
+import {
+  AppAbility,
+  createAppAbility,
+  defineRulesForPurchases,
+  User,
+} from './ability'
 import {SanityDocument} from '@sanity/client'
+import {subject} from '@casl/ability'
+
+test('can edit profile if current user matches profile', () => {
+  const user = {
+    id: 'abcdedf',
+    purchases: [],
+  }
+  const rules = defineRulesForPurchases({user})
+  const ability = createAppAbility(rules)
+
+  expect(
+    ability.can(
+      'edit',
+      subject('User', {
+        id: 'abcdedf',
+        purchases: [],
+      }),
+    ),
+  ).toBe(true)
+})
```

Comment:
this ability isn't used yet, but would be a way to enforce rules against specific subjects. It was breaking when I tried it so I thought this would make good documentation of how to perform this

we probably want to utilize this style for verifying access to specific content based on purchases in the future